### PR TITLE
Fully implemented sorting behavior for tools

### PR DESCRIPTION
**Pull request by haneenalsewari**
Issue #44 

**What Was Done:**
Refined sorting behavior for the tools catalog:
- The clicked tool is now consistently moved to the top (index 0).
- All other tools shift one position to the right (n + 1).
- The top three tools are consistently marked with "Recently Used" badges.
- Fixed inconsistencies where some tools weren't properly reordered after multiple clicks.
- Re-implemented logic to apply sorting changes to all tools, not just the recently used ones.

### CODE REVIEW

File: web/js/playground.js [hunk 1]
```diff
@@ -33,22 +33,16 @@ function saveRecentlyUsedTools(recentTools) {
...
 
     // Remove the tool if it exists in recentTools, then add it to the top
     recentTools = recentTools.filter(tool => tool.label.toLowerCase() !== clickedTool.label.toLowerCase());
     recentTools.unshift(clickedTool);
 
-    const topThreeTools = recentTools.slice(0, 3);
-    saveRecentlyUsedTools(topThreeTools);
-    console.log("updated recent tools after click:", topThreeTools);
+    //const topThreeTools = recentTools.slice(0, 3);
+    saveRecentlyUsedTools(recentTools.slice(0, 3));
```
Comment (sesheppard) [inline web/js/playground.js:44]: I suppose this is a way to keep track of the top tools, if it works as expected.

File: web/js/playground.js [hunk 2]
```diff
@@ -101,33 +95,15 @@ function renderTools() {
...
-    // Sort recently used tools based on their order in recentTools array
-    const sortedRecentlyUsedTools = recentlyUsedTools.sort(
-        (a, b) =>
-            recentToolsSet.has(a.label.toLowerCase()) - recentToolsSet.has(b.label.toLowerCase())
-    );
-
-    // Combine sorted recently used tools and non-recently-used tools
-    const sortedTools = [...sortedRecentlyUsedTools, ...nonRecentlyUsedTools];
+    const sortedTools = [...recentTools, ...ToolsCatalog.filter(tool => 
+        !recentTools.some(recentTool => recentTool.label === tool.label))];
```
Comment (sesheppard) [inline web/js/playground.js:101]: I can only find where the list of recent tools is passed between functions within the code, and not where it is initially defined. What exactly moves a tool to the top of the list when it is used?

**Review by alar12 — CHANGES_REQUESTED** [review]
Ensure that when a tool is used it moves to the top and the previous tool moves to the second position, and so on.

**Comment by alar12** [comment]
The tools are not displayed on the page when the local storage is empty.

**Review by alar12 — APPROVED** [review]

**Pull request merged**